### PR TITLE
Default levels

### PR DESCRIPTION
**Pull request by ashih2018**
Created method for 'set_default_levels' where preexisting levels (level 0-4) are created and saved to the data base for the rubric criterion, with default names and descriptions. 
Possible shortcomings: 
- Currently, two of the tests relating to the create_and_update_from_csv_row are failing. I am currently working on that method on a branch that is branched off of this one. We may want to wait until I finish that method so that all the tests work and I can merge that branch into this one and this one into sti-criteria so that all the tests pass?
- I used rubocop -a to lint all three files. However, this resulted in me linting beyond what I worked on. Some of the autolint resulted in changing some lines of code, and I don't know whether these should be changed back or not.

### CODE REVIEW

File: app/models/level.rb [hunk 1]
```diff
@@ -1,13 +1,10 @@
 # Level represent a level within a Rubric Criterion
 class Level < ApplicationRecord
-  belongs_to :rubric_criterion
```
Comment (david-yz-liu) [inline app/models/level.rb]: I'm not sure why you removed this, this is pretty important.

File: app/models/rubric_criterion.rb [hunk 2]
```diff
@@ -34,31 +36,35 @@ def update_assigned_groups_count
...
-     'description' => I18n.t('rubric_criteria.defaults.description_1')},
-    {'name' => I18n.t('rubric_criteria.defaults.level_2'),
-     'description' => I18n.t('rubric_criteria.defaults.description_2')},
-    {'name' => I18n.t('rubric_criteria.defaults.level_3'),
-     'description' => I18n.t('rubric_criteria.defaults.description_3')},
-    {'name' => I18n.t('rubric_criteria.defaults.level_4'),
-     'description' => I18n.t('rubric_criteria.defaults.description_4')}
-  ]
+
+  def levels
```
Comment (david-yz-liu) [inline app/models/rubric_criterion.rb]: Remove this---you can access `self.levels` already, just by having the `has_many` association defined above.

File: spec/models/rubric_criterion_spec.rb [hunk 3]
```diff
@@ -33,13 +33,15 @@
...
-      r.set_default_levels
-      expect(r.level_0_name).to eq(I18n.t('rubric_criteria.defaults.level_0'))
-      expect(r.level_1_name).to eq(I18n.t('rubric_criteria.defaults.level_1'))
-      expect(r.level_2_name).to eq(I18n.t('rubric_criteria.defaults.level_2'))
-      expect(r.level_3_name).to eq(I18n.t('rubric_criteria.defaults.level_3'))
-      expect(r.level_4_name).to eq(I18n.t('rubric_criteria.defaults.level_4'))
+      assignment = create(:assignment)
+      rubric = create(:rubric_criterion, assignment: assignment)
+      rubric.set_default_levels
+      levels = rubric.get_levels
```
Comment (david-yz-liu) [inline spec/models/rubric_criterion_spec.rb]: You should just do `rubric.levels` here.

File: app/models/rubric_criterion.rb [hunk 4]
```diff
@@ -12,6 +12,8 @@ class RubricCriterion < Criterion
 
   has_many :tas, through: :criterion_ta_associations
 
+  has_many :levels
```
Comment (david-yz-liu) [inline app/models/rubric_criterion.rb]: This is good; add a default order scope on the level number (see https://stackoverflow.com/questions/18284606/deprecated-warning-for-rails-4-has-many-with-order for details).

File: spec/models/rubric_criterion_spec.rb [hunk 5]
```diff
@@ -223,7 +225,7 @@
 
         context 'be able to create a new instance with level descriptions' do
           it 'not raise error' do
-            criterion = RubricCriterion.create_or_update_from_csv_row(@csv_base_row, @assignment)
+            criterion = RubricCriterion.create_or_update_fbyerom_csv_row(@csv_base_row, @assignment)
```
Comment (david-yz-liu) [inline spec/models/rubric_criterion_spec.rb]: Make sure to revert this change.

File: app/models/rubric_criterion.rb (hunk 4 above)
Comment (ashih2018) [reply app/models/rubric_criterion.rb]: should I order on level number or level mark?
